tensorflow/core/kernels/linalg/lu_op_gpu.cu.cc

```diff
@@ -182,9 +182,15 @@ class LuOpGpu : public AsyncOpKernel {
       auto packed_triangular_factors_ptrs = solver->GetScratchSpace<uint8>(
           sizeof(Scalar*) * batch_size, "packed_triangular_factors_ptrs",
           /* on_host */ true);
+#if GOOGLE_CUDA
       const Scalar** packed_triangular_factors_ptrs_base =
           reinterpret_cast<const Scalar**>(
               packed_triangular_factors_ptrs.mutable_data());
+#else //TENSORFLOW_USE_ROCM
+      Scalar** packed_triangular_factors_ptrs_base =
```

Comment:
Can we have both as const, or both as non-const?

Reply:
We do this in a few other places (here for one: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/linalg/matrix_solve_op.cc#L230-L236)
Let me get to the bottom on why this is this way for ROCm.